packages/zoe/test/unitTests/contracts/test-sellTickets.js

```diff
@@ -0,0 +1,567 @@
+
+    const jokerInvite = await E(ticketSalesPublicAPI).makeBuyerInvite();
+
+    const {
+      terms: { pricePerItem },
+    } = await E(zoe).getInstanceRecord(ticketSalesInstanceHandle);
+
+    // Joker does NOT check available tickets and tries to buy the ticket
+    // number 1(already bought by Alice, but he doesn't know)
+    const ticket1Amount = ticketAmountMath.make(
+      harden([
+        {
+          show: 'Steven Universe, the Opera',
+          start: 'Wed, March 25th 2020 at 8pm',
+          number: 1,
+        },
+      ]),
+    );
+
+    const jokerProposal = harden({
+      give: { Money: pricePerItem },
+      want: { Items: ticket1Amount },
+    });
+
+    const jokerPaymentForTicket = jokerPurse.withdraw(pricePerItem);
+
+    const { outcome, payout: payoutP } = await zoe.offer(
+      jokerInvite,
+      jokerProposal,
+      harden({
+        Money: jokerPaymentForTicket,
+      }),
+    );
+
+    t.rejects(
+      outcome,
+      /Some of the wanted items were not available for sale/,
+      'ticket 1 is no longer available',
+    );
+
+    const payout = await payoutP;
+    const jokerTicketPayoutAmount = await ticketIssuer.getAmountOf(
+      payout.Items,
+    );
+    const jokerMoneyPayoutAmount = await moolaIssuer.getAmountOf(payout.Money);
+
+    t.ok(
+      ticketAmountMath.isEmpty(jokerTicketPayoutAmount),
+      'Joker should not receive ticket #1',
+    );
+    t.deepEquals(
+      jokerMoneyPayoutAmount,
+      moola(22),
+      'Joker should get a refund after trying to get ticket #1',
+    );
+  };
+
+  // Joker attempts to buy ticket 2 for 1 moola (and should fail)
+  const jokerTriesToBuyTicket2 = async (
+    ticketSalesInstanceHandle,
+    moola100Payment,
+  ) => {
+    const { publicAPI: ticketSalesPublicAPI } = await E(zoe).getInstanceRecord(
+      ticketSalesInstanceHandle,
+    );
+    const ticketIssuer = await E(ticketSalesPublicAPI).getItemsIssuer();
+    const ticketAmountMath = await E(ticketIssuer).getAmountMath();
+
+    const jokerPurse = await E(moolaIssuer).makeEmptyPurse();
+    await E(jokerPurse).deposit(moola100Payment);
+
+    const jokerInvite = await E(ticketSalesPublicAPI).makeBuyerInvite();
+
+    const ticket2Amount = ticketAmountMath.make(
+      harden([
+        {
+          show: 'Steven Universe, the Opera',
+          start: 'Wed, March 25th 2020 at 8pm',
+          number: 2,
+        },
+      ]),
+    );
+
+    const insufficientAmount = moola(1);
+    const jokerProposal = harden({
+      give: { Money: insufficientAmount },
+      want: { Items: ticket2Amount },
+    });
+
+    const jokerInsufficientPaymentForTicket = jokerPurse.withdraw(
+      insufficientAmount,
+    );
+
+    const { outcome, payout: payoutP } = await zoe.offer(
+      jokerInvite,
+      jokerProposal,
+      harden({
+        Money: jokerInsufficientPaymentForTicket,
+      }),
+    );
+
+    t.rejects(
+      outcome,
+      /(?<=More money)(.*)(?=is required to buy these items)/,
```

Comment:
I had to do some research to learn that these are non-capturing look-ahead and look-behind. I think this is not useful here. The test is only using the outcome of the match, so the capturing group, and the look-ahead and look-behind aren't necessary.  `/More money.*is required to buy these items/,` is simpler and clearer.

Reply:
Sure, that sounds good. Thanks for the research!